backend/backend/authentication.py

```
@@ -0,0 +1,172 @@
+
+        try:
+            # Try to get the user from the local database
+            user = User.objects.get(firebase_uid=uid)
+            return user
+        except User.DoesNotExist:
+            # User not found locally, proceed to create
+            logger.info(f"User with firebase_uid {uid} not found locally. Attempting creation.")
+            user_details = self._fetch_user_details(uid, decoded_token)
+            return self._create_user_with_profile(uid, user_details)
+
+    def _fetch_user_details(self, uid, decoded_token):
```

Ruff (0.8.2)

Missing return type annotation for private function `_fetch_user_details`

(ANN202)